capnp/src/primitive_list.rs

```diff
@@ -71,6 +71,15 @@ impl <'a, T: PrimitiveElement> Reader<'a, T> {
         assert!(index < self.len());
         PrimitiveElement::get(&self.reader, index)
     }
+
+    #[cfg(all(target_endian = "little"))]
+    pub fn to_slice(&self) -> &[T] {
+        let bytes = self.reader.into_raw_bytes();
+        unsafe {
+            use std::slice;
+            slice::from_raw_parts(bytes.as_ptr() as *mut T, (data_bits_per_element(T::element_size())/8) as usize)
```

Comment:
It's possible that `self.reader.get_element_size()` is larger than `T::element_size()`. See https://capnproto.org/language.html#evolving-your-protocol:
> A field of type `List(T)`, where `T` is a primitive type, blob, or list, may be changed to type `List(U)`, where `U` is a struct type whose `@0` field is of type T.

In such a case, this function will return bad data.

Reply:
Also, the second argument to `slice::from_raw_parts()` needs to use `bytes.len()` somehow. 
(Looks like the `Builder` version of `to_slice()` got this correct.)

Reply:
I made it return an option, with something if the underlying data was still of the right number of bits or nothing if it's migrated.